refactor: restructure frontend architecture

HijoDeHonor opened the pull request:
Separated larger components into smaller ones, better distributed logic, and implemented a service for future use with useContext.

HijoDeHonor committed:
refactor: restructure frontend architecture

Separated larger components into smaller ones, better distributed logic, and implemented a service for future use with useContext.

HijoDeHonor committed:
refactor: restructure frontend architecture

Separated larger components into smaller ones, better distributed logic, and implemented a service for future use with useContext.

HijoDeHonor committed:
Merge branch 'refactor/2-frontend-architecture' of https://github.com/PiewDev/ia-workout-fe into refactor/2-frontend-architecture

HijoDeHonor committed:
refactor: update import paths, create button component with CSS, rebase from develop

Updated import paths for some components. Created a button component with its respective CSS for standard implementations. Rebases performed from the develop branch.

PiewDev commented on index.html:
```diff
@@ -4,7 +4,7 @@
     <link rel="icon" type="image/svg+xml" href="/vite.svg" />
     <meta name="viewport" content="width=device-width, initial-scale=1.0" />
-    <title>Vite + React</title>
+    <title>DistopicWorkout</title>
```
```suggestion
    <title>distopicWorkout</title>
```

PiewDev commented on src/pages/questionaire/Questionaire/Questionnaire.css:
```diff
@@ -0,0 +1,119 @@
+  font-weight: bold;
+  text-align: center;
+}
+
```
remove all css that is not in use like .next-button

PiewDev commented on src/pages/questionaire/Questionaire/Questionnaire.css:
```diff
@@ -0,0 +1,119 @@
+  margin-bottom: 20px;
+}
+
+h2 {
```
move h2-h3 and others 'primitives' types to a class

PiewDev commented on src/pages/routine/Routine/Routine.jsx:
```diff
@@ -0,0 +1,33 @@
+      <p>Tipo de plan: {plan.planType}</p>
+      <p>Días por semana: {plan.daysPeerWeek}</p>
+      
+      <button onClick={() => setShowAbout(!showAbout)}>
```
change button to Button

PiewDev reviewed (changes requested):
There are some changes to make, but above all implement that all buttons are the new Button component, there is no button that is different in visual and functionality.

ngrandoli committed:
refactor: rename files and folders as per QA standards, move CSS to respective components

Renamed several files and folders to follow QA-required naming conventions. Moved parts of the CSS to their respective components.

PiewDev commented on src/components/landing-page/LandingPage.jsx:
```diff
@@ -1,5 +1,6 @@
 import React from 'react';
 import './LandingPage.css';
+import Button from '../button/Button';
```
```suggestion
import Button from '../button/Button.jsx';
```

PiewDev commented on src/components/landing-page/LandingPage.jsx:
```diff
@@ -20,12 +21,7 @@ const LandingPage = () => {
-        >
-          Crear Rutina
-        </button>
+        <Button onClick={handleCreateRoutine} isDisabled={false} />
```
text is missing

PiewDev commented on src/pages/questionaire/components/text-input-question/TextInputQuestion.jsx:
```diff
@@ -0,0 +1,15 @@
+        onChange={(e) => onInput(e.target.value)}
+        value={currentAnswer || ''}
+        className="text-input"
+        placeholder={`Máximo ${limit} caracteres`}
```
move to text constants

PiewDev commented on src/pages/routine/Routine.jsx:
move text  to text constants

PiewDev commented on src/pages/routine/components/week/week.jsx:
move text to text constants

PiewDev commented on src/services/questions/getQuestions.js:
The error messages are in different languages

HijoDeHonor committed:
change the bouncing for an actual spinner

HijoDeHonor committed:
refactor: replace strings with constants, update filenames with -tmp suffix, align error message with language used

Replaced strings with constants to improve maintainability. Added the suffix `-tmp` to filenames to handle case sensitivity issues in the repository. Updated an error message to align with the language used.

HijoDeHonor committed:
fix: remove -tmp suffix to resolve case sensitivity issues

Removed the `-tmp` suffix from filenames to fix case sensitivity problems in the repository.

HijoDeHonor committed:
refactor: replace strings with constants.
Replaced strings with constants to improve maintainability.
